File: frontend/src/domain/audio/value-objects/audio-format.ts
/**
 * Audio format value object.
 */

export enum AudioFormat {
    WAV = 'wav',
    AAC = 'aac',
    PCM = 'pcm',
    MP3 = 'mp3',
    FLAC = 'flac',
}

export function getAudioFormatExtension(format: AudioFormat): string {
    return `.${format}`;
}

export function isLosslessFormat(format: AudioFormat): boolean {
    return format === AudioFormat.WAV || format === AudioFormat.PCM || format === AudioFormat.FLAC;
}

File: backend/streamcraft/domain/audio/value_objects/audio_format.py
"""Audio format value object."""

from enum import Enum


class AudioFormat(str, Enum):
    """Audio file format."""

    WAV = "wav"
    AAC = "aac"
    PCM = "pcm"
    MP3 = "mp3"
    FLAC = "flac"

    def __str__(self) -> str:
        """String representation."""
        return self.value

    @property
    def extension(self) -> str:
        """Get file extension."""
        return f".{self.value}"

    @property
    def is_lossless(self) -> bool:
        """Check if format is lossless."""
        return self in (AudioFormat.WAV, AudioFormat.PCM, AudioFormat.FLAC)
